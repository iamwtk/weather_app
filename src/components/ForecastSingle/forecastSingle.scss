@import "../../scss/variables";
@import "../../scss/mixins";

.forecast_single {
  color: $white;
  text-align: center;
  border-right: 1px solid rgba(255,255,255,.5);
  border-bottom: 1px solid rgba(255,255,255,.5);
  width: 50%;
  padding: 0.25em;

  &:nth-child(2n+0) {
    border-right: none;
  }

  &:nth-child(3),
  &:nth-child(4) {
    border-bottom: none;
  }
  @include sm {
    width: 25%;
    border-bottom: none;
    border-right: 1px solid rgba(255,255,255,.5);

    &:nth-child(2n+0) {
      border-right: 1px solid rgba(255,255,255,.5);
    }

    &:last-child {
      border-right: none;
    }
  };

  .day {
    font-family: $font-book;
    font-size: 1em;
    @include lg {
      font-size: 1.75em;
    };
  }

  .weather_icon {
    width: 40px;
    height: auto;
    display: block;
    margin: 0.5em auto;

    path {
      fill: $white;
    }
    @include lg {
      width: 100px;
      margin: 0.25em auto;
    };
  }

  .temp {
    font-family: $font-book;
    font-size: 1.5em;
    @include lg {
      font-size: 2em;
    };
  }
}
