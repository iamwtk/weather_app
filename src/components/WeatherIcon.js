import React, { Component } from 'react';

/**
 * Loading SVG icons as Component via svg-react-loader in webpack
 * @type {Component}
 */
import IconDay01 from '-!svg-react-loader?name=MyIcon!../icons/day01.svg'
import IconNight01 from '-!svg-react-loader?name=MyIcon!../icons/night01.svg'
import IconDay02 from '-!svg-react-loader?name=MyIcon!../icons/day02.svg'
import IconNight02 from '-!svg-react-loader?name=MyIcon!../icons/night02.svg'
import IconDay03 from '-!svg-react-loader?name=MyIcon!../icons/day03.svg'
import IconNight03 from '-!svg-react-loader?name=MyIcon!../icons/night03.svg'
import IconDay04 from '-!svg-react-loader?name=MyIcon!../icons/day04.svg'
import IconNight04 from '-!svg-react-loader?name=MyIcon!../icons/night04.svg'
import IconDay09 from '-!svg-react-loader?name=MyIcon!../icons/day09.svg'
import IconNight09 from '-!svg-react-loader?name=MyIcon!../icons/night09.svg'
import IconDay10 from '-!svg-react-loader?name=MyIcon!../icons/day10.svg'
import IconNight10 from '-!svg-react-loader?name=MyIcon!../icons/night10.svg'
import IconDay11 from '-!svg-react-loader?name=MyIcon!../icons/day11.svg'
import IconNight11 from '-!svg-react-loader?name=MyIcon!../icons/night11.svg'
import IconDay13 from '-!svg-react-loader?name=MyIcon!../icons/day13.svg'
import IconNight13 from '-!svg-react-loader?name=MyIcon!../icons/night13.svg'
import IconDay50 from '-!svg-react-loader?name=MyIcon!../icons/day50.svg'
import IconNight50 from '-!svg-react-loader?name=MyIcon!../icons/night50.svg'

class WeatherIcon extends Component {

  constructor(props) {
    super(props)
    /**
     * Assigns SVG icon modules to icon codes received from OpenWeatherMap API
     * @type {Object}
     */
    this.icons = {
      '01d': IconDay01,
      '01n': IconNight01,
      '02d': IconDay02,
      '02n': IconNight02,
      '03d': IconDay03,
      '03n': IconNight03,
      '04d': IconDay04,
      '04n': IconNight04,
      '09d': IconDay09,
      '09n': IconNight09,
      '10d': IconDay10,
      '10n': IconNight10,
      '11d': IconDay11,
      '11n': IconNight11,
      '13d': IconDay13,
      '13n': IconNight13,
      '50d': IconDay50,
      '50n': IconNight50
    }
  }

  render() {
    /**
     * Gets Component from this.icons Object by key passed as a prop to WeatherIcon Component
     * then renders Component
     * if icon was not passed defaults to IconDay1
     * @type {Component}
     */
    let IconName;
    console.log(this.props);
    this.props.icon ?
      IconName = this.icons[this.props.icon] :
      IconName = IconDay01



    return <IconName className={this.props.addClass ? this.props.addClass : 'null'}/>

  }

}

export default WeatherIcon;
