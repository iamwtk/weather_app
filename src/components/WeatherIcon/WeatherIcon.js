import React, { Component } from 'react';

/**
 * Loading SVG icons as Component via svg-react-loader in webpack
 * @type ReactComponent
 */
import IconDay01 from '-!svg-react-loader?name=MyIcon!../../icons/day01.svg'
import IconNight01 from '-!svg-react-loader?name=MyIcon!../../icons/night01.svg'
import IconDay02 from '-!svg-react-loader?name=MyIcon!../../icons/day02.svg'
import IconNight02 from '-!svg-react-loader?name=MyIcon!../../icons/night02.svg'
import IconDay03 from '-!svg-react-loader?name=MyIcon!../../icons/day03.svg'
import IconNight03 from '-!svg-react-loader?name=MyIcon!../../icons/night03.svg'
import IconDay04 from '-!svg-react-loader?name=MyIcon!../../icons/day04.svg'
import IconNight04 from '-!svg-react-loader?name=MyIcon!../../icons/night04.svg'
import IconDay09 from '-!svg-react-loader?name=MyIcon!../../icons/day09.svg'
import IconNight09 from '-!svg-react-loader?name=MyIcon!../../icons/night09.svg'
import IconDay10 from '-!svg-react-loader?name=MyIcon!../../icons/day10.svg'
import IconNight10 from '-!svg-react-loader?name=MyIcon!../../icons/night10.svg'
import IconDay11 from '-!svg-react-loader?name=MyIcon!../../icons/day11.svg'
import IconNight11 from '-!svg-react-loader?name=MyIcon!../../icons/night11.svg'
import IconDay13 from '-!svg-react-loader?name=MyIcon!../../icons/day13.svg'
import IconNight13 from '-!svg-react-loader?name=MyIcon!../../icons/night13.svg'
import IconDay50 from '-!svg-react-loader?name=MyIcon!../../icons/day50.svg'
import IconNight50 from '-!svg-react-loader?name=MyIcon!../../icons/night50.svg'


/*
 * WeatherIcon Component
 * renders SVG icon based on icon code passed via props
 */
class WeatherIcon extends Component {
  /**
   * Sets icon component based on icon id string passed as props
   * @param  {String} icon string of icon id passed as props
   * @return {ReactComponent} ReactComponent matching icon id
   */
  getIconComponent(icon) {
    const icons = {
      '01d': IconDay01,
      '01n': IconNight01,
      '02d': IconDay02,
      '02n': IconNight02,
      '03d': IconDay03,
      '03n': IconNight03,
      '04d': IconDay04,
      '04n': IconNight04,
      '09d': IconDay09,
      '09n': IconNight09,
      '10d': IconDay10,
      '10n': IconNight10,
      '11d': IconDay11,
      '11n': IconNight11,
      '13d': IconDay13,
      '13n': IconNight13,
      '50d': IconDay50,
      '50n': IconNight50
    }
    return icons[icon]
  }
  /**
   * Component Render Method
   * @return ReactElement markup
   */
  render() {
    /**
     * Checks if icon id was passed as props
     * if yes calls getIconComponent with icon id as parameter
     * if not defaults to IconDay01
     * @type {ReactComponent}
     */
    let IconName
    this.props.icon ?
      IconName = this.getIconComponent(this.props.icon) :
      IconName = IconDay01

    return <IconName className={this.props.addClass ? this.props.addClass : 'null'}/>

  }

}

export default WeatherIcon
