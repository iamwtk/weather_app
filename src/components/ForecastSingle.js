import React from 'react'

import WeatherIcon from './WeatherIcon';

import style from './forecastSingle.scss';

const ForecastSingle = (props) => (
  <div className={style.forecast_single}>
    <span key={props.data.day} className={style.day}>{props.data.day}</span>
    <WeatherIcon icon={props.data.icon} key={props.data.icon} addClass={style.weather_icon}/>
    <span key={props.data.temp} className={style.temp}>{props.data.temp}&#176;</span>
  </div>
);

export default ForecastSingle
