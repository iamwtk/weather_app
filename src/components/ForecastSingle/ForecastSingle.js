import React from 'react'

import WeatherIcon from '../WeatherIcon/WeatherIcon'

import style from './forecastSingle.scss'

/**
 * ForecastSingle React component
 * Renders single widget for daily forecast
 * @param {object} props object with data to display
 */
const ForecastSingle = (props) => {
  const { day, icon, temp } = props.data
  return (
    <div className={style.forecast_single}>
      <span key={day} className={style.day}>{day}</span>
      <WeatherIcon icon={icon} key={icon} addClass={style.weather_icon}/>
      <span key={temp} className={style.temp}>{temp}&#176;</span>
    </div>
  )
}
export default ForecastSingle
