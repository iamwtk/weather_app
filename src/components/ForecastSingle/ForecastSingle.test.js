import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ForecastSingle from './ForecastSingle'
import WeatherIcon from '../WeatherIcon/WeatherIcon';
import style from './forecastSingle.scss'


describe('<ForecastSingle />', () => {
  let component
  const data = {
    day: 'Monday',
    icon: '05n',
    temp: '23'
  }
  
  beforeEach(() => {
    component = shallow(<ForecastSingle data={data}/>)
  })

  describe('Snapshot', () => {
    it('should match the snapshot', () => {
      const tree = toJson(component)
      expect(tree).toMatchSnapshot()
    })
  })
})
