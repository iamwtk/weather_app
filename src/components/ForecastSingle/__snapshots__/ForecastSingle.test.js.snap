// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastSingle /> Snapshot should match the snapshot 1`] = `
<div
  className="forecast_single"
>
  <span
    className="day"
    key="Monday"
  >
    Monday
  </span>
  <WeatherIcon
    addClass="weather_icon"
    icon="05n"
    key="05n"
  />
  <span
    className="temp"
    key="23"
  >
    23
    Â°
  </span>
</div>
`;
